gateway-contracts/contracts/Decryption.sol

```diff
@@ -295,11 +318,26 @@ contract Decryption is
         bytes[] storage verifiedSignatures = $._verifiedPublicDecryptSignatures[decryptionId][digest];
         verifiedSignatures.push(signature);
 
-        /// @dev Send the event if and only if the consensus is reached in the current response call.
-        /// @dev This means a "late" response will not be reverted, just ignored
+        // Store the KMS transaction sender address for the public decryption response
+        // It is important to consider the same mapping fields used for the consensus
+        // A "late" valid KMS transaction sender address will still be added in the list.
+        $.consensusTxSenderAddresses[decryptionId][digest].push(msg.sender);
+
+        // Send the event if and only if the consensus is reached in the current response call.
+        // This means a "late" response will not be reverted, just ignored and no event will be emitted
         if (!$.decryptionDone[decryptionId] && _isConsensusReachedPublic(verifiedSignatures.length)) {
             $.decryptionDone[decryptionId] = true;
 
+            // A "late" valid KMS could still see its transaction sender address be added to the list
+            // after consensus. This storage variable is here to be able to retrieve this list later
+            // by only knowing the decryption ID, since a consensus can only happen once per decryption
+            // request, independently of the decryption response type (public or user).
+            $.decryptionConsensusDigest[decryptionId] = digest;
+
+            // Store the decrypted result for the public decryption response in order to be able to
+            // retrieve it later
+            $.publicDecryptedResult[decryptionId] = decryptedResult;
+
             emit PublicDecryptionResponse(decryptionId, decryptedResult, verifiedSignatures);
```

Comment:
Same question here, or is this gas optimization on the events for another issue?